### DISCUSSION TITLE
Officially support Wayand

### DISCUSSION THREAD

**Discussion by KayJay7** [Floorp-Projects/Floorp#1114] [Feature Request]
As stated in a comment [!1063](https://github.com/Floorp-Projects/Floorp/issues/1063#issuecomment-2030969722) Floorp doesn't currently support Wayland, as a consequence issues related to Wayland might be dismissed as won't fix. While using Xwayland apparently works without issues, running natively on Wayland would be preferred.

Knowing that:
1. Firefox supports Wayland upstream
2. Xorg is not being actively developed anymore, just maintained
3. Many major distros (including ubuntu, the most used one) are shifting toward Wayland
4. Some Firefox features are Wayland only, and cannot be implemented on X11 due to its limitations

Wayland support should be reconsidered.

As a matter of facts, most issues with Wayland will be or are already fixed upstream, and that specific issue (which happens to be the only major issue related to Wayland that I know of in Floorp) is already solved, as reported in [!1065](https://github.com/Floorp-Projects/Floorp/discussions/1065), and will probably be included as part of the next ESR anyway (in July).

In the end, this feature request is mostly a statement against hiding behind the lame "Wayland is unsupported" excuse, while there is no reason for it to be unsupported.

**Comment by CutterKnife** [comment]
Thank you for your interest in Floorp!

Floorp basically follows Firefox.
Changing the Firefox code may increase the scope of what needs to be maintained or introduce unintended bugs. They may make exceptions if there are many requests, but please be aware that ignoring Firefox's implementation is not really what they want. Some of the changes can be seen [here](https://github.com/Floorp-Projects/Floorp-core/blob/main/docs/changed-firefox-codes.md).
Once this is implemented in the upstream ESR, it will be implemented in Floorp.

Pull requests on this are always welcome :)
